RUN3/AliAnalysisTaskAO2Dconverter.cxx

```diff
@@ -1904,39 +1906,96 @@ void AliAnalysisTaskAO2Dconverter::FillEventInTF()
   } // end loop on calo cells
   eventextra.fNentries[kCalo] = ncalocells_filled;
 
+  
+  // Trigger data for EMCAL:
+  // - For full payload (monitoring) events store all non-0 L1 ADCs
+  // - For regular events store non-0 L1 ADCs of the 3 leading Gamma patches
   AliEMCALGeometry *geo = AliEMCALGeometry::GetInstanceFromRunNumber(fVEvent->GetRunNumber()); // Needed for EMCAL trigger mapping
   AliVCaloTrigger *calotriggers = fVEvent->GetCaloTrigger("EMCAL");
+  Bool_t fullPayload = gRandom->Uniform() < fFractionL1MonitorEventsEMCAL;
+  calotrigger.fIndexBCs = fBCCount;
+  calotrigger.fFastOrAbsID = 10001;
+  calotrigger.fLnAmplitude = fullPayload ? 1 : 0;
+  calotrigger.fTriggerBits = 0;
+  calotrigger.fCaloType = 1;
+  FillTree(kCaloTrigger);
```

Comment:
I think you need to increase ncalotriggers_filled on the FillTree calls

Reply:
It is done below. The first 3 entries are global event properties which are mapped with unique FastOR Abs IDs to be interpreted as status codes. Consequently I start ncalotriggers_filled with 3.